### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by smiasojed**

### CODE REVIEW

File: contracts/solidity/openzeppelin_solang.patch [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ }
+diff --git a/contracts/utils/Strings.sol b/contracts/utils/Strings.sol
+index 657ebd6..1e8ec52 100644
+--- a/contracts/utils/Strings.sol
++++ b/contracts/utils/Strings.sol
+@@ -29,7 +29,7 @@ library Strings {
+                 ptr--;
+                 /// @solidity memory-safe-assembly
+                 assembly {
+-                    mstore8(ptr, byte(mod(value, 10), _SYMBOLS))
```
[the remark this answers is not in the document]
Comment (smiasojed) [reply contracts/solidity/openzeppelin_solang.patch:97]: commented out all block

File: contracts/solidity/openzeppelin_solang.patch [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+             }
+diff --git a/contracts/utils/Address.sol b/contracts/utils/Address.sol
+index 0791a66..f0dc85a 100644
+--- a/contracts/utils/Address.sol
++++ b/contracts/utils/Address.sol
+@@ -42,7 +42,8 @@ library Address {
+         // for contracts in construction, since the code is only stored at the end
+         // of the constructor execution.
+ 
+-        return account.code.length > 0;
```
[the remark this answers is not in the document]
Comment (smiasojed) [reply contracts/solidity/openzeppelin_solang.patch:37]: added return is_contract(account);

File: contracts/solidity/openzeppelin_solang.patch [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
++    ) internal returns (bytes memory) {
++        (bool success, bytes memory returndata) = target.call(data);
+         return verifyCallResultFromTarget(target, success, returndata, errorMessage);
+     }
+ 
+@@ -182,7 +183,7 @@ library Address {
+         bytes memory data,
+         string memory errorMessage
+     ) internal returns (bytes memory) {
+-        (bool success, bytes memory returndata) = target.delegatecall(data);
```
[the remark this answers is not in the document]
Comment (smiasojed) [reply contracts/solidity/openzeppelin_solang.patch]: fixed

File: contracts/solidity/openzeppelin_solang.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (smiasojed) [reply contracts/solidity/openzeppelin_solang.patch:107]: fixed

File: contracts/solidity/openzeppelin_solang.patch [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
++++ b/contracts/token/ERC721/ERC721.sol
+@@ -409,10 +409,10 @@ contract ERC721 is Context, ERC165, IERC721, IERC721Metadata {
+                 if (reason.length == 0) {
+                     revert("ERC721: transfer to non ERC721Receiver implementer");
+                 } else {
+-                    /// @solidity memory-safe-assembly
+-                    assembly {
+-                        revert(add(32, reason), mload(reason))
+-                    }
++                  //  /// @solidity memory-safe-assembly
```
[the remark this answers is not in the document]
Comment (smiasojed) [reply contracts/solidity/openzeppelin_solang.patch:13]: It does not fail in our benchmark test

**Review by xermicus — APPROVED** [review]

**Pull request merged**